codec-dns/src/main/java/io/netty/handler/codec/dns/TcpDnsResponseEncoder.java

```diff
@@ -0,0 +1,81 @@
+package io.netty.handler.codec.dns;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToMessageEncoder;
+
+import java.util.List;
+
+@ChannelHandler.Sharable
+public final class TcpDnsResponseEncoder extends MessageToMessageEncoder<DnsResponse> {
+    private final DnsRecordEncoder encoder;
+
+    public TcpDnsResponseEncoder() {
+        this(DnsRecordEncoder.DEFAULT);
+    }
+
+    public TcpDnsResponseEncoder(DnsRecordEncoder encoder) {
+        this.encoder = encoder;
+    }
+
+    @Override
+    protected void encode(ChannelHandlerContext ctx, DnsResponse response, List<Object> out) throws Exception {
+        ByteBuf buf = ctx.alloc().ioBuffer(1024);
+        boolean success = false;
+        try {
+            buf.writerIndex(buf.writerIndex() + 2);
+            encodeHeader(response, buf);
+            this.encodeQuestions(response, buf);
+            this.encodeRecords(response, DnsSection.ANSWER, buf);
+            this.encodeRecords(response, DnsSection.AUTHORITY, buf);
+            this.encodeRecords(response, DnsSection.ADDITIONAL, buf);
```

Comment:
nit: remove `this.` (for all method calls above)

Reply:
removed all the `this.`